storage/kv.go

```diff
@@ -38,6 +42,17 @@ type KV interface {
 	TxnPut(txnID int64, key, value []byte) (rev int64, err error)
 	TxnDeleteRange(txnID int64, key, end []byte) (n, rev int64, err error)
 
+	// Watcher watches the events happening or happened in etcd. The whole
```

Comment:
That sounds to abstract it better. It may cause KV to expose some new API to facilitate watch in the future or even this PR in my design. Will try.